src/test/ui/parser/issue-68987-unmatch-issue-3.stderr

```diff
@@ -0,0 +1,21 @@
+error: unexpected closing delimiter: `}`
+  --> $DIR/issue-68987-unmatch-issue-3.rs:7:1
+   |
+LL | fn f(i: u32, j: u32) {
+   |                      - this delimiter might not be properly closed...
+...
+LL |     }
+   |     - ...as it matches this but it has different indentation
+LL | }
+   | ^ unexpected closing delimiter
+
+error: mismatched closing delimiter: `)`
+  --> $DIR/issue-68987-unmatch-issue-3.rs:4:19
+   |
+LL |     while cnt < j {
+   |                   ^ unclosed delimiter
+LL |         write!&mut res, " ");
+   |                            ^ mismatched closing delimiter
```

Comment:
Interesting. This feels like something we could handle: the second error is way more relevant and we could maybe do a couple of passes to see if solving that one "fixes" the other one ("did someone swap delimiter types or simply forgot to open them?"). *not for this PR necessarily*

Reply:
Yes, somehow like what we have done for functions's arugments checking, we may improve with more trivial suggestions like minimal edit distance algorithm in future.

Reply:
We should move the second error to first, seems when mismatching error happens, also comes with unclosing delimiter errors, but mismatching error is more releated to root cause.

Reply:
@estebank we merge this PR first, then coming with another one? 
Seems it's better to review.

Reply:
Ping @estebank , any more suggestion on this PR?